atsign-foundation/at_demos
# feat: at_notifications Dart Demo

**- What I did**
- New `at_notifications` Dart SDK demo
- Two main programs: `subscribe.dart` and `send-notification.dart`
- **NEW** +1 `clear-notifications.dart` - clears the notifications in your `notify:list`
- README showing usage and example

**- How I did it**

**- How to verify it**

Getting Started section of `README.md`:

1. To run this demo, you will need 2 atSigns and thier .atKeys in `~/.atsign/keys` directory.

2. Simply start your subscriber with the following command:

```sh
dart run subscribe.dart -a @subscriber -r ".*@notifier"
```

Example output:

```
Authenticated as @subscriber
Subscribed to notifications with regex ".*"
```

3. Run the notify program to send a notification to the subscriber:

```sh
dart run send-notification.dart -f @notifier -t @subscriber -m "hello, world"
```

You should see a similar output in the subscriber's terminal:

```sh
[NOTIFICATION RECEIVED] =>
    id: 2704e281-c882-4ab2-8cb9-6c54fba538f5
    key: @soccer99:message.at_notifications_demo@soccer0
    from: @soccer0
    to: @soccer99
    epochMillis: 1686525575956
    status: 
    value: poopy
    operation: update
    messageType: MessageType.key
    isEncrypted: true
    expiresAtInEpochMillis: null
    metadata: Metadata{ttl: null, ttb: null, ttr: null,ccd: null, isPublic: false, isHidden: false, availableAt : null, expiresAt : null, refreshAt : null, createdAt : null, updatedAt : null, isBinary : false, isEncrypted : null, isCached : false, dataSignature: null, sharedKeyStatus: null, encryptedSharedKey: null, pubKeyCheckSum: null, encoding: null, encKeyName: null, encAlgo: null, ivNonce: 65NeBNKzEqksyeTS1Wbm+g==, skeEncKeyName: null, skeEncAlgo: null}
```

**- Description for the changelog**
